lms/services/jstor.py

```diff
@@ -28,17 +47,58 @@ def via_url(self, request, document_url):
         """
 
         return via_url(
-            request,
-            document_url,
-            content_type="pdf",
-            options={"via.jstor.site_code": self._site_code},
+            request, document_url=self._get_public_url(document_url), content_type="pdf"
+        )
+
+    def _get_public_url(self, url):
+        """
+        Get a signed S3 URL for the given JSTOR URL.
+
+        :param url: The URL to stream
+        :return: A public URL
+        :raises ExternalRequestError: If we get bad data back from the service
+        """
+
+        doi = url.replace("jstor://", "")
+        if "/" not in doi:
+            doi = f"{self.DEFAULT_DOI_PREFIX}/{doi}"
+
+        token = self._get_access_token(self._site_code)
+        s3_url = self._http.request(
+            method="GET",
+            url=f"{self._api_url}/pdf-url/{doi}",
+            headers={"Authorization": f"Bearer {token}"},
+        ).text
+
+        if not s3_url.startswith("https://"):
+            raise ExternalRequestError(
+                f"Expected to get an S3 URL but got: '{s3_url}' instead"
+            )
+
+        return s3_url
+
+    def _get_access_token(self, site_code):
+        return jwt.encode(
```

Comment:
This could be done using our `JWTService.encode_with_secret`
```python
encode_with_secret({"site_code": site_code},  self._secret, timedelta(hours=1))


Is not abstracting much but it will make the transaction to jwt-jose easier (just switching it on the service).
```

Reply:
Yep. This wasn't around in Via, I'll fix that up